install.php

```diff
@@ -6,99 +6,89 @@
+    JFile::delete($path . '/' . $filename);
 
-		$filename = 'civimembershiptypes.php';
-		echo '<p>Removing file from: '.$path.'/'.$filename.'</p>';
+    $filename = 'civimembershiptypes.php';
+    echo '<p>Removing file from: ' . $path . '/' . $filename . '</p>';
 
-    JFile::delete($path.'/'.$filename);
+    JFile::delete($path . '/' . $filename);
 
-		echo '<p>Important! Failure to enable an authentication module will likely result in your being locked out of your site!</p>';
+    echo '<p>Important! Failure to enable an authentication module will likely result in your being locked out of your site!</p>';
 
-	}
- 
-	/**
-	 * method to update the plugin
-	 *
-	 * @return void
-	 */
-	function update($parent) 
-	{
-		// $parent is the class calling this method
-		// echo '<p>' . JText::_('PLG_CIVICRM_UPDATE_TEXT') . '</p>';
-	}
- 
-	/**
-	 * method to run before an install/update/uninstall method
-	 *
-	 * @return void
-	 */
-	function preflight($type, $parent) 
-	{
-		// $parent is the class calling this method
-		// $type is the type of change (install, update or discover_install)
-	}
- 
-	/**
-	 * method to run after an install/update/uninstall method
-	 *
-	 * @return void
-	 */
-	function postflight($type, $parent) 
-	{
-		// $parent is the class calling this method
-		// $type is the type of change (install, update or discover_install)
-		// File move is in post flight as we have to wait on the installer having installed it first.
-		//
-		if ($type == "install") {
+  }
 
-	    $path1 = JPATH_SITE.'/plugins/authentication/civicrm';
-			$path2 = JPATH_SITE.'/administrator/components/com_civicrm/civicrm/joomla/site/elements';
+  /**
+   * method to update the plugin
+   *
+   * @return void
+   */
+  function update($parent) {
+    // $parent is the class calling this method
+    // echo '<p>' . JText::_('PLG_CIVICRM_UPDATE_TEXT') . '</p>';
+  }
 
-			$filename = 'civimembershiplevels.php';
-			echo '<p>Move from: '.$path1.'/'.$filename.'</p>';
-			echo '<p>To: '.$path2.'/'.$filename.'</p>';
-			JFile::copy($path1.'/'.$filename, $path2.'/'.$filename);
+  /**
+   * method to run before an install/update/uninstall method
+   *
+   * @return void
+   */
+  function preflight($type, $parent) {
+    // $parent is the class calling this method
+    // $type is the type of change (install, update or discover_install)
+  }
 
-			$filename = 'civimembershiptypes.php';
-			echo '<p>Move from: '.$path1.'/'.$filename.'</p>';
-			echo '<p>To: '.$path2.'/'.$filename.'</p>';
-			JFile::copy($path1.'/'.$filename, $path2.'/'.$filename);
+  /**
+   * method to run after an install/update/uninstall method
+   *
+   * @return void
+   */
+  function postflight($type, $parent) {
+    // $parent is the class calling this method
+    // $type is the type of change (install, update or discover_install)
+    // File move is in post flight as we have to wait on the installer having installed it first.
+    //
+    if ($type == "install") {
```

Comment:
I agree. update this to include both install and update (us in_array)

Reply:
ok